### PR TITLE
CP-26405: Ported build from oasis to jbuilder.

### PR DESCRIPTION
**Pull request by kc284**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,45 +1,29 @@
...
-setup.data:
-	$(SETUP) -configure $(CONFIGUREFLAGS)
+# requires odoc
+doc:
+	jbuilder build @doc
 
-configure:
-	$(SETUP) -configure $(CONFIGUREFLAGS)
+reindent:
+	git ls-files '*.ml*' '**/*.ml*' | xargs ocp-indent --syntax cstruct -i
```
Comment (mseri) [inline Makefile]: ` '*.ml*'` is enough, `ls-files` looks for the files recursively in the folders

File: lib/jbuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+(library (
+  (name tapctl)
+  (public_name xapi-tapctl)
+  (wrapped false)
+  (flags (:standard -w -37-39 %s))
+  (libraries (
+    rpclib
+    threads
+    forkexec
```
Comment (mseri) [inline lib/jbuild]: Looking at `lib/tapctl.ml` it seems that we should also add `xapi_stdext_unix`, `xapi_stdext_std`, `xapi_stdext_threads` and `xapi_stdext_monadic` here

File: lib/tapctl.ml [hunk 3]
```diff
@@ -1,450 +1,451 @@
...
 let tap_ctl = canonicalise "tap-ctl"
 
 let invoke_tap_ctl ctx cmd args =
-	let find x = try [ x ^ "=" ^ (Sys.getenv x) ] with _ -> [] in
-	let env = Array.of_list (find "PATH" @ (find "TAPDISK") @ (find "TAPDISK2")) in
-	let stdout, stderr = execute_command_get_output ~env tap_ctl (cmd::args) in
-	stdout
+  let find x = try [ x ^ "=" ^ (Sys.getenv x) ] with _ -> [] in
+  let env = Array.of_list (find "PATH" @ (find "TAPDISK") @ (find "TAPDISK2")) in
+  let stdout, stderr = execute_command_get_output ~env tap_ctl (cmd::args) in
```
Comment (mseri) [inline lib/tapctl.ml]: I think this could be replaced by `Unix.open_process_in` as in 
```ocaml
let read_process_lines command =
  let in_channel = Unix.open_process_in command in
  let rec fill lines =
    try
        let line = input_line in_channel in
        fill line::lines
    with End_of_file ->
      ignore (Unix.close_process_in in_channel);
      lines
  in
  List.rev lines
```
and drop the dependency on `forkexec` but probably this should be a separate PR

**Comment by mseri** [comment]
Does this require a spec file update?

File: lib/tapctl.ml (hunk 3 above)
Comment (kc284) [reply lib/tapctl.ml]: It belongs to a different PR :)

**Comment by kc284** [comment]
Yes, it does.

File: xapi-tapctl.opam [hunk 4]
```diff
@@ -1,24 +1,18 @@
...
 depends: [
-  "ocamlfind" {build}
-  "oasis" {build}
-  "xapi-stdext" {>= "2.0.0"}
+  "jbuilder" {build}
+  "xapi-stdext-unix"
+  "xapi-stdext-std"
+  "xapi-stdext-threads"
+  "xapi-stdext-monadic"
   "xapi-forkexecd"
```
Comment (mseri) [inline xapi-tapctl.opam]: I think `"forkexec"` is enough for this package, but whoever used it will need the daemons as well, so `xapi-forkexecd` could be fine as well

Comment (mseri) [reply xapi-tapctl.opam]: Oh, we are dropping the dependency... than let's not waste time on it :)
